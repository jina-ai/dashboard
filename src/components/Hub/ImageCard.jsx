import React from "react";
import { Row, Col, Card } from "react-bootstrap";
import { Link } from "react-router-dom";
import StarRating from "./StarRating";

export default function ImageCard({ image }) {
  let {
    name,
    id,
    official,
    author,
    description,
    totalStars,
    totalRatings,
    numReviews,
  } = image;

  let rating;
  if (image.totalStars && image.totalRatings) {
    rating = totalStars / totalRatings;
  }

  let reviews;
  if (!numReviews) reviews = <span className="text-muted">no reviews</span>;
  else if (numReviews > 1) reviews = <span>{numReviews} reviews</span>;
  else reviews = <span>{numReviews} review</span>;

  return (
    <Link className="unstyled-link" to={`/package?id=${id}`}>
      <Card className="clickable mb-4 h-100">
        <Card.Body className="pb-0 mb-0 pt-3">
          <Row>
            <Col xs="12" className="px-0">
              <div className="app-title">
                {name}
                {official && (
                  <span title="Official Package" className="float-right">
                    <i className="ml-2 material-icons verified-icon">
                      verified_user
                    </i>
                  </span>
                )}
              </div>
              <div className="app-subtitle">{author}</div>
            </Col>
            <Col sm="12" className="px-0 pt-3 pb-0">
              <div className="description-container">
                <div className="description-overlay" />
                <div className="app-description">{description}</div>
              </div>
            </Col>
          </Row>
        </Card.Body>
        <Card.Footer className="pt-0 px-3 pb-3">
          {reviews}
          {rating && <StarRating rating={rating} />}
        </Card.Footer>
      </Card>
    </Link>
  );
}
