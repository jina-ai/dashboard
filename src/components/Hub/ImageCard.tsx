import React from "react"
import { Row, Col, Card } from "react-bootstrap"
import styled from "@emotion/styled"
import { Link } from "react-router-dom"

type HubImagePreview = {
  name: string
  author: string
  description: string
  kind: string
  keywords: string[]
}

type Props = {
  image: HubImagePreview
  index: number
}

export const Tag = styled.div`
  background: ${(props) => props.theme.palette.tagBackground};
  border-radius: 6px;
  display: inline-block;
  padding: 0.25rem;
  margin-right: 1rem;
  margin-bottom: 1rem;
  white-space: nowrap;
`
const Title = styled.div`
  font-size: 1.25em;
  font-weight: 700;
  line-height: normal;
`

const SubTitle = styled.div`
  font-weight: 600;
  opacity: 0.5;
`

const ImageLink = styled(Link)`
  color: unset;
  &:hover {
    text-decoration: none;
  }
`

export default function ImageCard({ image, index }: Props) {
  let { name, author, keywords, kind, description } = image

  return (
    <ImageLink to={`/package/${kind}/${index}`}>
      <Card className="clickable mb-4 h-100 image-card" data-name="hubImage">
        <Card.Body className="pb-0 mb-0 pt-3">
          <Row>
            <Col xs="12" className="px-0">
              <Title className="mb-2">{name}</Title>
              {keywords.map((keyword, index) => (
                <Tag data-name="hubImageTags" key={index}>
                  {keyword}
                </Tag>
              ))}
              <SubTitle data-name="hubImageAuthor" className="mb-2">
                {author}
              </SubTitle>
            </Col>
            <Col sm="12" className="px-0 pb-0">
              <div className="description-container">
                <div className="description-overlay" />
                <div className="app-description">{description}</div>
              </div>
            </Col>
          </Row>
        </Card.Body>
      </Card>
    </ImageLink>
  )
}
