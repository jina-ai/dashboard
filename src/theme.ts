const PERSIAN_GREEN = "#009999"
const GOLDEN_TANOI = "#FFCC66"
const BITTER_SWEET = "#ff6666"
const SCOOTER_GREEN = "#32C8CD"
const WHITE = "#FFFFFF"
const DADGREY = "#DAD7FE"
const SKY_DIVING_IN_THE_BLUE = "#007AFF"
const QUICK_SILVER = "#A6A6A6"
const HONEYBEE = "#FFCC66"
const BLACK = "#000000"

const theme = {
  palette: {
    background: WHITE,
    headerTextColor: BLACK,
    primary: PERSIAN_GREEN,
    secondary: GOLDEN_TANOI,
    warning: GOLDEN_TANOI,
    error: BITTER_SWEET,
    success: SCOOTER_GREEN,
    info: PERSIAN_GREEN,
    critical: "#ff4540",
    debug: "#7E7278",
    tagBackground: DADGREY,
    tagPlatformBackground: HONEYBEE,
    mutedText: QUICK_SILVER,
    highlight: SKY_DIVING_IN_THE_BLUE,
  },
}

export { theme }
