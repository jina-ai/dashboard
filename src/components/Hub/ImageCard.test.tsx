import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "@emotion/react"
import { theme } from "../../theme"
import ImageCard from "./ImageCard"
import { HashRouter as Router } from "react-router-dom"

test("ImageCard", () => {
  const image = {
    name: "Enel",
    author: "Oda",
    keywords: ["encoder"],
    kind: "encoder",
    description: "whatever",
  }
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Router>
        <ImageCard image={image} index={7} />
      </Router>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
