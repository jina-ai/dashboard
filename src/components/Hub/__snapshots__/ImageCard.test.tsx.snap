// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard 1`] = `
<a
  className="css-sirbk0"
  href="#/package/encoder/7"
  onClick={[Function]}
>
  <div
    className="clickable mb-4 h-100 image-card card"
    data-name="hubImage"
  >
    <div
      className="pb-0 mb-0 pt-3 card-body"
    >
      <div
        className="row"
      >
        <div
          className="px-0 col-12"
        >
          <div
            className="mb-2 css-14i9fhd"
          >
            Enel
          </div>
          <div
            className="css-1gs0ali"
            data-name="hubImageTags"
          >
            encoder
          </div>
          <div
            className="mb-2 css-k133o5"
            data-name="hubImageAuthor"
          >
            Oda
          </div>
        </div>
        <div
          className="px-0 pb-0 col-sm-12"
        >
          <div
            className="description-container"
          >
            <div
              className="description-overlay"
            />
            <div
              className="app-description"
            >
              whatever
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;
